### PR TITLE
Adding checkpoint for persisting event log state between restarts

### PR DESCRIPTION
**Pull request by andrewkroh**
`Checkpoint` persists event log state information to disk so that event log monitoring can resume from the last read event in the case of a restart or unexpected interruption.

### CODE REVIEW

File: beat/winlogbeat.go [hunk 1]
```diff
@@ -13,8 +14,12 @@ import (
 	"github.com/elastic/libbeat/common"
 	"github.com/elastic/libbeat/logp"
 	"github.com/elastic/libbeat/publisher"
+	"github.com/elastic/winlogbeat/checkpoint"
 	"github.com/elastic/winlogbeat/config"
 	"github.com/elastic/winlogbeat/eventlog"
+
+	// XXX: Remove before releasing.
```
Comment (ruflin) [inline beat/winlogbeat.go]: Can you open an issue for that? Otherwise we probably forget.

File: checkpoint/checkpoint.go [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+	}
+
+	logp.Debug("checkpoint", "Checkpoint saved to disk. numPending=%d",
+		c.numUpdates)
+	c.numUpdates = 0
+	return true
+}
+
+// flush writes the current state to disk.
+func (c *Checkpoint) flush() error {
```
Comment (ruflin) [inline checkpoint/checkpoint.go]: Could we reuse the code from the registrar here?

**Comment by ruflin** [comment]
LGTM. I thinking if we should "abstract" out the registrar from filebeat so both use the same registrar and we iterate on top of it?

File: beat/winlogbeat.go (hunk 1 above)
Comment (andrewkroh) [inline beat/winlogbeat.go]: I removed this line. It is easy enough to add back for testing purposes.

**Comment by andrewkroh** [comment]
I did spend some time looking at the registrar from Filebeat to see how I could reuse the code. I struggled to come up with a way to reuse the code because the types `FileState` and `EventLogState` were specific to the beat (also I wanted to tweak the behavior a bit: json vs yaml, write to disk on each event vs batching a few events before writing).

I could create a more generic implementation that uses `interface{}` and then each beat embeds this this type and wraps all access to the data in methods that do type checking. I would be happy to try this out after we merge all beats to one repo (that makes it a lot easier IMO).

**Comment by ruflin** [comment]
@andrewkroh +1 on this.

**Pull request merged**